### DISCUSSION TITLE
是否可以增加更多的部署方式？（Can more deployment be added?）

### DISCUSSION THREAD

**Discussion by asasn** [usememos/memos#2813] [General]
这毫无疑问是个非常好用的工具，但对Windows的机器就不怎么友好了，因为还需要安装一个docker。
最新版的docker对win10版本有要求，看似简单，实则连系统都得更新。
如果能够提供更加便利的部署方式就好了。
我刚刚看了一下，有人打包了其他版本的，但那些都是第三方，安全性没有保障不说，版本更新也未必及时。
（This is undoubtedly a very useful tool, but it is not so friendly to Windows machines, because you also need to install a docker.
The latest version of docker has requirements for the win10 version, which seems simple, but in fact even the system has to be updated.
It would be nice to provide a more convenient way to deploy.
I just looked at it, and some people have packaged other versions, but those are third parties, and the security is not guaranteed, and the version update may not be timely.）

**Comment by jkjoy** [comment]
直接编译运行

**Reply by asasn** [reply]
> 直接编译运行

你好，我现在尝试着自行编译……但卡在go的那一步了。
按照这个教程 https://www.cnblogs.com/CDF5BAC6/p/18217607 来的，看来是网络问题……有什么不kx上网的方式解决吗？比如说别人编译好的main.exe……能不能用？能用的话，去哪里找？
- [x]  buf-Windows-x86_64.exe generate
- [x]  corepack enable && pnpm i --frozen-lockfile
- [x]  pnpm build
- [ ] go build main.go

**Reply by jkjoy** [reply]
[memos.zip](https://github.com/user-attachments/files/16921135/memos.zip)
我给你打包一个,我有点好奇你不科学上网也能github?

**Reply by asasn** [reply]
> [memos.zip](https://github.com/user-attachments/files/16921135/memos.zip) 我给你打包一个,我有点好奇你不科学上网也能github?

多谢大佬！！

我在发帖之前成功啦！go env -w GOPROXY=https://goproxy.cn,direct 原来如此简单，改了一下代理。

是的，GitHub可以直连啊。只不过有时候会抽风。要等一会儿才好。

现在公网ip，内网穿透这些也搞好了，下一步琢磨云服务器。哈哈

**Comment by lincolnthalles** [comment]
#### See
* [Memospot](https://github.com/lincolnthalles/memospot): lightweight Memos Windows/Linux/macOS client with the server built-in.
* [memos-builds](https://github.com/lincolnthalles/memos-builds): Memos binary builds and Docker containers for **_all_** supported platforms.
* [Windows service guide](https://github.com/lincolnthalles/memos-builds/blob/main/docs/windows-service.md): if you have an always-on Windows Server machine.

Docker Desktop has a huge memory overhead (>2GB) as it runs inside a virtual machine. It's simply not worth it if you only want to run Memos. Memospot will use about 500 MB of RAM, and this includes the WebView used to display its interface. 

#### About releases
The official Memos Docker releases follow a rolling release/continuous delivery model, which allows repeatedly patching the same version number. Docker image pulling still detects it as proper upgrades via hashing, enabling a faster release cycle and a simpler CI/CD pipeline.

Although it's possible to release official binaries right now, there are some challenges involved and it would increase the complexity of release cycles, and that's probably why there are no official binaries yet.

Releasing binaries means that all and any patches must have at least a new patch version for end-user control, allowing for better tracking and reproduction of issues. However, this can also mean slower release cycles and more thorough testing, and Memos hasn't even reached version 1.0.0 yet.

To illustrate that matter a little: occasionally, on the [memos-builds](https://github.com/lincolnthalles/memos-builds) repository, to cope with the current rolling release model, I have to delete a release and recreate it with updated upstream Memos code that fixed some issues, as it would be troublesome if I just upgraded the version numbering by myself. That's why it's hard to keep 1:1 releases with the upstream project.

The trust matter is up to you, but what you get at the memos-builds repository (or container registry) is built on GitHub-hosted runners with unaltered upstream code.

**Reply by asasn** [reply]
Thank you very much. Trying[Memospot](https://github.com/lincolnthalles/memospot)
